README.md

```diff
@@ -2,142 +2,295 @@
-      name: 'Simon Grüber'
-    }
-  ]
+      _key: 'conclusion',
+      _type: 'paragraph',
+      text: 'Learning Sanity will help you take control of your content workflow.',
+    },
+  ],
 }
 
-// Specify id if the two documents do not match
-const operations = diffPatch(itemA, itemB, {id: itemA._id, ifRevisionID: itemA._rev})
-await sanityClient.transaction(operations).commit()
-
-// Patches generated:
-const generatedPatches = [
-  {
-    patch: {
-      id: 'die-hard-iii',
-      ifRevisionID: 'k0k0s',
-      set: {
-        'name': 'Die Hard with a Vengeance',
-        'characters[_key=="l13ma92"].name': 'Simon Grüber'
-      },
-    }
-  },
-  {
-    patch: {
-      id: 'die-hard-iii',
-      unset: ['year']
-    }
-  }
+// Generate patches that capture each user's intent
+const patchA = diffPatch(originalDoc, userAChanges)
+const patchB = diffPatch(originalDoc, userBChanges)
+
+// Apply both patches - they merge successfully because they target different aspects
+// User A's reordering and typo fix + User B's content improvement both apply
+const finalMergedResult = {
+  _id: 'blog-post-123',
+  _type: 'blogPost',
+  title: 'Getting Started with Sanity',
+  paragraphs: [
+    {
+      _key: 'intro',
+      _type: 'paragraph',
+      text: 'Sanity is a complete content operating system that gives developers control over the entire stack.', // ✅ User B's improvement
+    },
+    {
+      _key: 'conclusion', // ✅ User A's reordering
+      _type: 'paragraph',
+      text: 'Learning Sanity will help you take control of your content workflow.',
+    },
+    {
+      _key: 'benefits',
+      _type: 'paragraph',
+      text: 'It offers real-time collaboration and gives developers control over the entire stack.', // ✅ User A's typo fix
+    },
+  ],
 }
 ```
 
-## diff-match-patch
+## Technical Details
 
-When encountering two strings to compare, this module will (by default) attempt to use [diff-match-patch (DMP)](https://www.sanity.io/docs/http-patches#diffmatchpatch-aTbJhlAJ) to transition the string from one state to the next.
+### String Diffing with diff-match-patch
 
-While these patches are extremely helpful (especially in a real-time, collaborative environment), they sometimes grow quite large and can be hard for humans to read. There are a few options you can use to tweak _when_ this module will use these patches, and when to fall back to a regular `set`-patch instead.
+When comparing strings, the library attempts to use [diff-match-patch](https://www.sanity.io/docs/http-patches#diffmatchpatch-aTbJhlAJ) to generate granular text patches instead of simple replacements. This preserves editing intent and enables better conflict resolution.
 
-The default rules says:
+**Automatic selection criteria:**
 
-- If the target string is below 30 characters long, don't use DMP
-- If the generated DMP is greater than 1.2 times larger than the target string, don't use DMP
+- **Document size limit**: Documents larger than 1MB use `set` operations
```

Comment:
Why does it matter how large the document is? Shouldn't this be based on the length of the string fields?

Reply:
Reading #36, this _is_ actually about string fields. I don't think we should use "document" in the name here. String size limit/string field size limit, maybe?